vernac/comTactic.mli

```diff
@@ -9,12 +9,14 @@
 (************************************************************************)
 
 (** Tactic interpreters have to register their interpretation function *)
-type 'a tactic_interpreter
-type interpretable = I : 'a tactic_interpreter * 'a -> interpretable
+type interpretable
 
-(** ['a] should be marshallable if ever used with [par:] *)
+(** ['a] should be marshallable if ever used with [par:]. Must be
+    called no more than once per process with a particular string: make
+    sure to use partial application. *)
```

Comment:
Reading the comment, I realized that this is a footgun. Can you wrap the closure inside a record type to enforce this statically?

Reply:
That doesn't enforce anything though.
There haven't been issues with CWarnings.create or Goptions.declare_bool_option_and_ref AFAIK so I'm not worried.

Reply:
Actually, I'd really prefer to keep the currently exported API in this file and add a function `eval : 'a tactic_interpreter -> 'a -> interpretable`.

Reply:
Such a function doesn't do any evaluation though, it just tuples the arguments.

Reply:
Whatever the name, the thing is that I believe it's a bad idea to export the type as a transparent function. BTW, I don't like the CWarnings / Goptions API either. For warnings, I am always confused by what is what and I have to copy paste snippets from other parts of the source.

Reply:
OK how about now

Reply:
BTW, I don't think one really needs a record, even a singleton would do
```ocaml
let Registered f = register_tactic_interpreter "name" f;;
```

I concur with @ppedrot that we can do better here.

Reply:
>let Registered f = register_tactic_interpreter "name" f;;

That's pretty good, I'll do that.